### DISCUSSION TITLE
Webserver won't start

### DISCUSSION THREAD

**Discussion by Cyphon00** [rdudhagra/Sand-Table#5] [Troubleshooting] [answered]
### Congratulations on a fantastic build and detailed instructions!

Could you please help me to get the webserver running.

I've followed the [Raspberry Pi instructions](https://github.com/rdudhagra/Sand-Table/tree/master/Build%20Guide#raspberry-pi-setupwebserver-installation), but the webserver won't start.

The output of `npm run prod` (run within /Website):
![image](https://user-images.githubusercontent.com/3205742/106139396-7ab1ff00-61c1-11eb-8a41-b0525f5d14fe.png)
![image](https://user-images.githubusercontent.com/3205742/106139532-a8974380-61c1-11eb-97c8-7fc7792f5646.png)

`npm install` produced many WARNINGS but no ERRORS.

Everything else up to that point seemed to install correctly.
Thanks for your help.

ps. I get the same results before and after installing nginx.

**Comment by rdudhagra** [comment, marked as answer]
Ah, that's a problem on my end. Basically the web server can't find a folder called `files` and panics.

Easy fix: Create a folder named `files` in the same directory as the folders src and server.

I'll add a more robust fix to the code shortly so that the server will just make this folder if it does not exist. Let me know if that solves your issue...

**Reply by rdudhagra** [reply]
Update: just fixed the code...just `git pull` again and the server should run

**Reply by Cyphon00** [reply]
Thanks very much! That fixed it.
I used the `git pull` method from `/Sand-Table`.

I do have a subsequent issue with **Upload a new track** unfortunately.
(Add new playlist worked fine however)

From the browser: 
![image](https://user-images.githubusercontent.com/3205742/106223664-fea5ce80-6235-11eb-90ac-996568a1c3c7.png)

From SSH
![image](https://user-images.githubusercontent.com/3205742/106222560-ed5bc280-6233-11eb-8770-028b1ffe0cb2.png)
![image](https://user-images.githubusercontent.com/3205742/106223753-22691480-6236-11eb-9774-f28505044600.png)


In case it's relevant, I'm still building the Arduino and controller board, so they're not connected:
>[Error: Error: No such file or directory, cannot open /dev/ttyACM0]
> [Error: Error: No such file or directory, cannot open /dev/ttyACM1]

**Reply by rdudhagra** [reply]
Hmm, interesting. I've never seen that before. To debug, would you mind sending over the contents of `dither_cells2.thr` and `dither_cells2.gcode`? Both should be in the `files` folder I mentioned earlier. My guess is that the issue is some sort of parsing error maybe from a comment or such that I'm not ignoring.

**Reply by Cyphon00** [reply]
Not a problem. Email sent!

**Reply by rdudhagra** [reply]
Note to others since this was handled offline: turns out the files were corrupted...no software bugs here!

**Reply by Cyphon00** [reply]
Thanks for your fast support!
